lib/services/role.go

```diff
@@ -1558,7 +1566,10 @@ func (set RoleSet) CheckAccessToServer(login string, s Server) error {
 	// the deny role set prohibits access.
 	for _, role := range set {
 		matchNamespace, namespaceMessage := MatchNamespace(role.GetNamespaces(Deny), s.GetNamespace())
-		matchLabels, labelsMessage := MatchLabels(role.GetNodeLabels(Deny), s.GetAllLabels())
+		matchLabels, labelsMessage, err := MatchLabels(role.GetNodeLabels(Deny), s.GetAllLabels())
+		if err != nil {
+			return trace.CompareFailed(err.Error())
```

Comment:
Is `trace.CompareFailed` checked anywhere? If not, I would just `trace.Wrap(err)` here as well.